### PR TITLE
🧪 [testing improvement] Boundary tests for _status_from_load

### PR DESCRIPTION
**Pull request by hu3mann**
The testing improvement addresses missing edge case tests for the `_status_from_load` function and its related utilities in the `ui-dashboard-backend` service.

🎯 **What:** The testing gap addressed was the lack of boundary value testing for the pure deterministic functions responsible for cognitive state status calculation.

📊 **Coverage:**
- `_status_from_load`: Tested at 0.0, 0.29, 0.3, 0.59, 0.6, 0.79, 0.8, 1.0, and out-of-range values (-0.5, 1.5).
- `_clamp`: Tested with values inside the range, below minimum, and above maximum.
- `_recommendation_for_status`: Tested for "low", "optimal", "high", "critical", and unknown status fallbacks.

✨ **Result:** Increased test coverage and confidence in the cognitive state status determination logic. The tests are deterministic and cover both happy paths and boundary conditions.

---
*PR created automatically by Jules for task [16729963237669949764](https://jules.google.com/task/16729963237669949764) started by @hu3mann*

### CODE REVIEW

File: tests/unit/test_ui_dashboard_logic.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+@pytest.fixture(scope="module")
+def dashboard_backend_module():
+    spec = importlib.util.spec_from_file_location("ui_dashboard_backend_app", MODULE_PATH)
+    assert spec and spec.loader
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module
+
+
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply tests/unit/test_ui_dashboard_logic.py:21]: Addressed in latest push.

File: tests/unit/test_ui_dashboard_logic.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    )
+    assert (
+        dashboard_backend_module._recommendation_for_status("critical")
+        == "Mandatory break now; switch to a low-complexity recovery task afterward."
+    )
+    # Test fallback
+    assert (
+        dashboard_backend_module._recommendation_for_status("unknown")
+        == "Continue current work patterns with short, planned breaks."
+    )
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply tests/unit/test_ui_dashboard_logic.py:76]: Addressed in latest push.

File: tests/unit/test_ui_dashboard_logic.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hu3mann) [reply tests/unit/test_ui_dashboard_logic.py:21]: Outdated thread auto-resolved after re-validation with no newer objections.

File: tests/unit/test_ui_dashboard_logic.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hu3mann) [reply tests/unit/test_ui_dashboard_logic.py:76]: Outdated thread auto-resolved after re-validation with no newer objections.
